File: Blueprints/BlueprintDefinitions/PreviewAspNetCoreWebAPI/test/ValuesControllerTests.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

using Xunit;
using Amazon.Lambda.Core;
using Amazon.Lambda.TestUtilities;
using Amazon.Lambda.APIGatewayEvents;

using Newtonsoft.Json;

using BLUEPRINT_BASE_NAME;


namespace BLUEPRINT_BASE_NAME.Tests
{
    public class ValuesControllerTests
    {


        [Fact]
        public async Task TestGet()
        {
            var lambdaFunction = new TestLambdaEntryPoint();

            var requestStr = File.ReadAllText("./SampleRequests/ValuesController-Get.json");
            var request = JsonConvert.DeserializeObject<APIGatewayProxyRequest>(requestStr);
            var context = new TestLambdaContext();
            var response = await lambdaFunction.FunctionHandlerAsync(request, context);

            Assert.Equal(response.StatusCode, 200);
            Assert.Equal("[\"value1\",\"value2\"]", response.Body);
            Assert.True(response.Headers.ContainsKey("Content-Type"));
            Assert.Equal("application/json; charset=utf-8", response.Headers["Content-Type"]);
        }


    }
}

File: Blueprints/BlueprintDefinitions/Msbuild-NETCore_2_0/AspNetCoreWebAPI-FSharp/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.fs
namespace BlueprintBaseName.1.Tests

open Xunit
open Amazon.Lambda.TestUtilities
open Amazon.Lambda.APIGatewayEvents
open System.IO
open Newtonsoft.Json
open BlueprintBaseName.1

module ValuesControllerTests =

    [<Fact>]
    let ``Request HTTP Get at /``() = async {
        let lambdaFunction = new LambdaEntryPoint()
        let requestStr = File.ReadAllText("./SampleRequests/ValuesController-Get.json")
        let request = JsonConvert.DeserializeObject<APIGatewayProxyRequest>(requestStr)
        let context = new TestLambdaContext()
        let! response = lambdaFunction.FunctionHandlerAsync(request, context) |> Async.AwaitTask

        Assert.Equal(200, response.StatusCode);
        Assert.Equal("[\"value1\",\"value2\"]", response.Body);
        Assert.True(response.Headers.ContainsKey("Content-Type"));
        Assert.Equal("application/json; charset=utf-8", response.Headers.Item("Content-Type"));
    }